pkg/pgmodel/pgx.go

```diff
@@ -282,8 +301,16 @@ func (p *pgxInserter) InsertData(rows map[string][]samplesInfo, ctx *InsertCtx)
 	} else {
 		go func() {
 			workFinished.Wait()
-			ctx.Close()
+			select {
+			case err = <-errChan:
+			default:
+			}
 			close(errChan)
+			if err != nil {
+				log.Error("msg", "error on send", "error", err)
+			} else if p.insertedDatapoints != nil {
```

Comment:
maybe make it clear in this message that this is in the async path and data is being lost